pkg/armhelpers/deploymentError.go

```diff
@@ -85,45 +72,36 @@ func DeployTemplateSync(az ACSEngineClient, logger *logrus.Entry, resourceGroupN
 
 	if depExt == nil {
 		logger.Warn("DeploymentExtended is nil")
-		return &apierror.Error{
-			Code:     apierror.InternalOperationError,
-			Message:  depErr.Error(),
-			Category: apierror.InternalError}
+		return &ArmError{
+			Code:    "DeploymentFailed",
+			Message: depErr.Error()}
 	}
 
 	// try to extract error from ARM Response
-	var armErr *apierror.Error
+	var armErr *ArmError
 	if depExt.Response.Response != nil && depExt.Body != nil {
 		buf := new(bytes.Buffer)
 		buf.ReadFrom(depExt.Body)
 		logger.Infof("StatusCode: %d, Error: %s", depExt.Response.StatusCode, buf.String())
 		if resp, err := toError(logger, buf.Bytes()); err == nil {
-			switch {
-			case depExt.Response.StatusCode < 500 && depExt.Response.StatusCode >= 400:
-				resp.Category = apierror.ClientError
-			case depExt.Response.StatusCode >= 500:
-				resp.Category = apierror.InternalError
-			}
 			armErr = resp
 		} else {
 			logger.Errorf("unable to unmarshal response into apierror: %v", err)
 		}
 	} else {
 		logger.Errorf("Got error from Azure SDK without response from ARM")
 		// This is the failed sdk validation before calling ARM path
-		return &apierror.Error{
-			Code:     apierror.InternalOperationError,
-			Message:  depErr.Error(),
-			Category: apierror.InternalError}
+		return &ArmError{
+			Code:    "DeploymentFailed",
+			Message: depErr.Error()}
 	}
 
-	// Check that ARM returned ErrorResponse
+	// Check that ARM returned an error
 	if armErr == nil || len(armErr.Message) == 0 || len(armErr.Code) == 0 {
-		logger.Warn("Not an ARM Response")
-		return &apierror.Error{
-			Code:     apierror.InternalOperationError,
-			Message:  depErr.Error(),
-			Category: apierror.InternalError}
+		logger.Warn("Not an ARM Error")
+		return &ArmError{
+			Code:    "DeploymentFailed",
```

Comment:
What is the difference between `ProvisioningFailed` and `DeploymentFailed`. Do we really care/need this distinction? Everything should be in `depErr.Error()`.

Reply:
Sure, we can use same error